.circleci/config.yml

```diff
@@ -187,3 +196,43 @@ jobs:
             echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
             docker push $IMAGE_NAME:$IMAGE_LATEST_TAG
             docker push $IMAGE_NAME:$IMAGE_VERSION_TAG
+  deploy-gar:
+    executor: gcp-gcr/default
+    parameters:
+      build_tag:
+        type: string
+        default: build
+      image:
+        type: string
+        default: firefoxprofiler
+      registry-url:
+        type: string
+        default: us-docker.pkg.dev
+    steps:
+      - checkout
+      - run:
+          name: Prepare environment variables for OIDC authentication
+          command: |
+            echo 'export GOOGLE_PROJECT_ID="moz-fx-profiler-prod"' >> "$BASH_ENV"
+            echo "export OIDC_WIP_ID=$GCPV2_WORKLOAD_IDENTITY_POOL_ID" >> "$BASH_ENV"
+            echo "export OIDC_WIP_PROVIDER_ID=$GCPV2_CIRCLECI_WORKLOAD_IDENTITY_PROVIDER" >> "$BASH_ENV"
+            echo "export GOOGLE_PROJECT_NUMBER=$GCPV2_WORKLOAD_IDENTITY_POOL_PROJECT_NUMBER" >> "$BASH_ENV"
+            echo "export OIDC_SERVICE_ACCOUNT_EMAIL=$GCP_SERVICE_ACCOUNT_EMAIL" >> "$BASH_ENV"
+            echo 'export GCP_GAR_REPO="profiler-prod"' >> "$BASH_ENV"
+            echo "export GAR_IMAGE=\"<<parameters.registry-url>>/${GOOGLE_PROJECT_ID}/${GCP_GAR_REPO}/<<parameters.image>>\"" >> $BASH_ENV
```

Comment:
For more simplicity, and if they are needed, I would create these env variables directly in the step below, like we do for `IMAGE_VERSION_TAG` `IMAGE_LATEST_TAG`.

For the hardcoded ones, it could be easier to add an `environment` property to the job or step like described in parts 2 and 3 in https://circleci.com/docs/set-environment-variable/.

Using the commands with `$BASH_ENV` are useful if:
* we want some interpolation
* we want these variables present for all the following steps

But I'm not opposed keeping them here if you prefer, buy only if we need them.